### PR TITLE
Adapting to new firewall API

### PR DESCRIPTION
**Pull request by schubi2**

### CODE REVIEW

File: src/include/dhcp-server/widgets.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 require "yast"
 require "ui/service_status"
+require "y2firewall/firewalld"
+require "y2firewall/helpers/interfaces"
```
Comment (teclator) [inline src/include/dhcp-server/widgets.rb]: if you include the helpers the `firewalld` method will be already defined and also you could use the interface_zone(name)` method

https://github.com/yast/yast-yast2/blob/master/library/network/src/lib/y2firewall/helpers/interfaces.rb#L55

File: src/include/dhcp-server/widgets.rb [hunk 2]
```diff
@@ -325,7 +326,7 @@ def OpenFirewallValidate(id, event)
         Builtins.foreach(@ifaces) do |ifcfg, interface|
           # interface is active
           if Ops.get_boolean(interface, "active", false) == true
-            if SuSEFirewall.GetZoneOfInterface(ifcfg) == nil
+            unless firewalld.zones.find { |z| z.interfaces.include?(ifcfg) }
```
Comment (teclator) [inline src/include/dhcp-server/widgets.rb]: np: I prefer any? 

Although would prefer to use `unless interface_zone(ifcfg)`

File: src/modules/DhcpServer.pm [hunk 3]
```diff
@@ -1514,11 +1514,11 @@ DHCP server will not be available."));
     @allowed_interfaces = split (/ /, $ifaces_list);
 
     # if firewall is enabled
-    if (SuSEFirewall->GetEnableService()) {
+    if (FirewalldWrapper->is_enabled()) {
 	foreach my $iface (@allowed_interfaces) {
-	    my $iface_zone = SuSEFirewall->GetZoneOfInterface($iface);
+	    my $iface_zone = FirewalldWrapper->zone_name_of_interface($iface);
```
Comment (teclator) [inline src/modules/DhcpServer.pm]: Has you confirmed that the default_zone is not used when no zone is returned? If the behavior is the same, then It is fine.

**Comment by teclator** [comment]
JFR: it depends on yast/yast-yast2#845

File: src/modules/DhcpServer.pm (hunk 3 above)
Comment (teclator) [reply src/modules/DhcpServer.pm]: Checked, seems it is the same behavior so LGTM

Comment (schubi2) [reply src/modules/DhcpServer.pm]: As far I see....yes

**Pull request merged**